code/modules/admin/verbs/warning.dm

```diff
@@ -334,6 +392,51 @@
 	usr << browse(dat, "window=lookupwarns;size=900x500")
 	feedback_add_details("admin_verb","WARN-LKUP")
 
+//Admin Proc to add a new User Notification
+/client/proc/notification_add()
+	set category = "Admin"
+	set name = "Add Notification"
+
+	if(!check_rights(R_ADMIN|R_MOD|R_DEBUG|R_DEV|R_CCIAA))
+		return
+
+	var/ckey = input(usr, "What ckey ?", "Enter a ckey")
+	if(!ckey)
+		to_chat(usr,"You need to specify a ckey.")
+		return
+
+	//Validate ckey
+	var/DBQuery/validatequery = dbcon.NewQuery("SELECT count(*) as count FROM ss13_player WHERE ckey = :ckey:")
+	validatequery.Execute(list("ckey" = ckey))
+	var/count = 0
+	if(!validatequery.NextRow())
+		to_chat(usr,"Could not find player with the specified ckey")
+		return
+	count = text2num(validatequery.item[1])
+	if(count != 1)
+		to_chat(usr,"Could not find player with the specified ckey")
+		return
+
+	var/list/types=list("player_greeting","player_greeting_chat","admin","ccia")
+	var/type = input(usr, "Which Type?", "Choose a type", "") as null|anything in (types)
+	if(!type)
+		to_chat(usr,"You need to specify a type.")
+		return
+
+	var/message = input(usr,"Notification Message", "Specify a notification message")
```

Comment:
Might be worth sanitizing the message with `sanitize(input(usr,"Notification Message", "Specify a notification message"))`